pkg/operator/ceph/controller/controller_utils.go

```diff
@@ -43,24 +42,24 @@ var (
 )
 
 // IsReadyToReconcile determines if a controller is ready to reconcile or not
-func IsReadyToReconcile(client client.Client, clustercontext *clusterd.Context, namespacedName types.NamespacedName, controllerName string) (cephv1.ClusterSpec, bool, bool, reconcile.Result) {
-	namespacedName.Name = namespacedName.Namespace
-	cephClusterExists := true
+func IsReadyToReconcile(c client.Client, clustercontext *clusterd.Context, namespacedName types.NamespacedName, controllerName string) (cephv1.ClusterSpec, bool, bool, reconcile.Result) {
```

Comment:
Maybe I was confused back then and somehow `namespacedName.Name` was empty but I think we should double-check. So the fix might be as simple as removing the line:

`namespacedName.Name = namespacedName.Namespace`

Reply:
Ah no, actually I remember now, the `namespacedName.Name` is the name of the watched resource so calling List sounds good to me!

Reply:
I actually started with that suggestion yesterday but then discovered the same. :)